<launch>
	<arg name="markerId"        default="47"/>
	<arg name="corr_markerId"   default="0"/>
	<arg name="corr_markerSize" default="0.1"/>    <!-- in m -->
	<arg name="markerSize"      default="0.05"/>    <!-- in m -->
	<arg name="eye"             default="left"/>
	<arg name="marker_frame"    default="aruco_marker_frame"/>
	<arg name="ref_frame"       default="world_frame_cam"/>  <!-- leave empty and the pose will be published wrt param parent_name -->
	<arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->

	<node pkg="aruco_ros" type="single_central" name="aruco_single_central" output="screen" >
		<remap from="/camera_info" to="/usb_cam/camera_info" />
		<remap from="/image" to="/usb_cam/image_rect" />
		<!-- <remap from="/image" to="/usb_cam/image_raw" /> -->
		<param name="image_is_rectified" value="True"/>
		<param name="marker_size"        value="$(arg markerSize)"/>
		<param name="marker_id"          value="$(arg markerId)"/>
		<param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
		<param name="camera_frame"       value="stereo_gazebo_$(arg eye)_camera_optical_frame"/>
		<param name="marker_frame"       value="$(arg marker_frame)" />
		<param name="corner_refinement"  value="$(arg corner_refinement)" />
	        <param name="which_team"         value="purple" />
	        <rosparam command="load" file="$(find aruco_ros)/config/cali_purple.yaml" />
        </node>

	<node pkg="aruco_ros" type="single_correction" name="aruco_single_correction">
		<remap from="/camera_info"      to="/usb_cam/camera_info" />
		<remap from="/image"            to="/usb_cam/image_rect" />
		<param name="image_is_rectified"        value="True" />
		<param name="marker_size"        value="$(arg corr_markerSize)"/>
		<param name="marker_id"          value="$(arg corr_markerId)"/>
		<param name="reference_frame"    value="$(arg ref_frame)"/>
		<param name="camera_frame"       value="stereo_gazebo_$(arg eye)_camera_optical_frame"/>
		<param name="marker_frame"       value="$(arg marker_frame)" />
		<param name="corner_refinement"  value="$(arg corner_refinement)" />
	</node>

	<node pkg="aruco_ros" type="dynamic_correction" name="camera_pose_correction" output="screen" >
                <param name="which_team"             value="purple" />
        </node>

	<node pkg="sample_pos" type="sample_pos.py" name="sample_pose" output="screen" />
</launch>
